### ISSUE TITLE
Support for Unicode characters

### ISSUE THREAD

**Issue by g-despot** [memgraph/memgraph#307]
## Description

Memgraph doesn't support Unicode characters. 
Even though nodes can contain such characters. functions don't work correctly with them.

![Imgur](https://i.imgur.com/4jGWhdN.png)

**Comment by antonio2368** [comment]
Yeah, unfortunately, our string functions don't work well with unicode strings. It's a tricky thing to deal with and for now, we didn't get a lot of requests for it. Your strings should be correctly stored but you cannot rely on any of the built-in string manipulations in Memgraph if your text contains any nonascii character.

Nevertheless, if you think something like this is really important we would love to hear it and take it into consideration.

**Comment by giorgiozoppi** [comment]
how much does costs to add libICU to the project?

**Comment by g-despot** [comment]
@giorgiozoppi are you also experiencing a problem because Unicode characters are not supported?

**Comment by gitbuda** [comment]
Related StackOverflow question -> https://stackoverflow.com/questions/73671795/how-to-do-an-order-by-ignoring-the-diacritical-marks-use-a-collation-in-cypher -> quick solution for some of the Unicode requirements would be to implement [magic functions](https://memgraph.com/docs/cypher-manual/functions#user-defined-memgraph-magic-functions).

**Comment by gitbuda** [comment]
+1 feature request here

**Comment by gitbuda** [comment]
It's actually possible to store Unicode characters, only the functions are now working properly. Import/export functionality might have issues, e.g. import_json.py doesn't work because the underlying file is not open with UTF encoding.

**Comment by DavIvek** [comment]
## Automated Triage Result: Not Applicable

This issue is a feature request and cannot be verified through automated reproduction testing.

---
*Automated triage by Memgraph Issue Triage Agent*
